src/java/htsjdk/samtools/SAMRecord.java

```diff
@@ -1665,7 +1807,12 @@ public int hashCode() {
             */
                 if (firstOnly) return ret;
             }
-            if (getHeader().getSequenceDictionary().size() == 0) {
+            if (getHeader() == null) {
+                if (ret == null) ret = new ArrayList<SAMValidationError>();
+                ret.add(new SAMValidationError(SAMValidationError.Type.MISSING_HEADER, "A non-null SAMHeader is required to resolve the sequence dictionary for a mapped read.", getReadName()));
+                if (firstOnly) return ret;
+            }
```

Comment:
We could go either way (before this change, there was already a validation error enum value for MISSING_HEADER, thoughits unclear to me if there was actually any code path that would produce such an error on a record with a null header without first throwing an NPE). One other option I like is to change the MISSING_HEADER enum value to be a SAMValidationError.Severity.WARNING instead of an ERROR, and only return those for each case where a header is required for validation (including SAMRecord.validateCigar and SAMUtils.validateCigar which can both be called independently of SAMRecord.isValid()).